src/v/raft/consensus.cc

```diff
@@ -2013,7 +2013,7 @@ consensus::do_append_entries(append_entries_request&& r) {
 
     // section 3
     if (request_metadata.prev_log_index < last_log_offset) {
-        if (unlikely(request_metadata.prev_log_index < _commit_index)) {
+        if (unlikely(request_metadata.prev_log_index < last_visible_index())) {
```

Comment:
Should we use `last_visible_index()` in line 2020 as well? (not entirely understanding its significance, but seems logical...)

Reply:
makes sense, i will change this